3rdparty/rnnoise-build/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+# Copyright The Mumble Developers. All rights reserved.
+# Use of this source code is governed by a BSD-style license
+# that can be found in the LICENSE file at the root of the
+# Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+include(FetchContent)
+
+set(RNNOISE_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../rnnoise-src")
+set(RNNOISE_MODEL_URL "https://media.xiph.org/rnnoise/models/rnnoise_data-")
+
+if(NOT EXISTS "${RNNOISE_SRC_DIR}/COPYING")
+	message(FATAL_ERROR
+		"${RNNOISE_SRC_DIR} was not found.\n"
+		"Please checkout the submodule:\n"
+		"git submodule update --init --recursive"
+	)
+endif()
+
+if(WIN32)
+	add_library(rnnoise SHARED)
+	set_target_properties(rnnoise PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+	if(MINGW)
+		# Remove "lib" prefix.
+		set_target_properties(rnnoise PROPERTIES PREFIX "")
+	endif()
+	target_compile_definitions(rnnoise
+		PRIVATE
+			"WIN32"
+			"DLL_EXPORT"
+	)
+else()
+	add_library(rnnoise STATIC)
+endif()
+
+target_compile_definitions(rnnoise PRIVATE "HAVE_CONFIG_H")
+
+target_include_directories(rnnoise
+	PRIVATE SYSTEM
+		${CMAKE_CURRENT_SOURCE_DIR}
+	PUBLIC SYSTEM
+		"${RNNOISE_SRC_DIR}/include"
+		"${RNNOISE_SRC_DIR}/src"
+)
+
+if(MSVC)
+	# Use malloc() and free() instead of variable length arrays (unsupported)
+	target_compile_definitions(rnnoise PRIVATE "USE_MALLOC")
```

Comment:
``USE_MALLOC`` is not present in upstream RNNoise. Davide introduced it in our first fork, because RNNoise was doing VLAs which it now apparently does not anymore. We can remove this then.

The ``_USE_MATH_DEFINES`` I'm not sure. It might be a MSVC specific. If it was instead also introduced for or first fork, please also remove it.